src/codegen/expression.rs

```diff
@@ -2766,14 +2796,35 @@ pub fn emit_function_call(
                 },
             );
 
-            vec![
+            let success = if ns.target.is_polkadot() {
+                let ret_code = Expression::Cast {
+                    loc: *loc,
+                    ty: Type::Uint(32),
+                    expr: Expression::Variable {
+                        loc: *loc,
+                        ty: Type::Bool,
+                        var_no: success,
+                    }
+                    .into(),
+                };
+                let ret_ok = Expression::NumberLiteral {
+                    loc: *loc,
+                    ty: Type::Uint(32),
+                    value: 0.into(),
+                };
+                Expression::Equal {
+                    loc: *loc,
+                    left: ret_code.into(),
+                    right: ret_ok.into(),
+                }
```

Comment:
So on the polkadot target, `success` is an u32. But since you said earlier that on Solana this variable isn't used anyways. I just turned the `success` var into an u32 instead. WDYT @LucasSte

Reply:
That's a good idea. I see in all other places (where you check the target before creating a temporary) you are creating the `success` variable as a boolean. Should those be `Type::Uint(32)` too?

Reply:
Well they might be all bools is because that's what you get on EVM for raw calls. But we don't support EVM for now, so I guess yes